scripts/text_classification/fasttext_word_ngram.py

```diff
@@ -239,10 +325,10 @@ def train(args):  # Load and clean data
     train_vocab = gluonnlp.Vocab(cntr)
```

Comment:
The current code does not really support n-gram vocab with n >= 2. In the paper, they use bins and hash for retrieving the embeddings.

Reply:
If you do not add n-grams to the vocab, they will all be mapped to the unknown token so all the n-grams share the same embedding features with unknown token.

Reply:
we first create token_indices as from vocab for the bigrams and then add these indices to both train and test set (used for evaluation)

        token_indices = {v: k + start_index for k, v in enumerate(ngram_set)}
        train_sequences = add_ngram(train_sequences, token_indices,
                                    ngram_range)
        test_sequences = add_ngram(test_sequences, token_indices, ngram_range)

Reply:
I do see that you add ngrams to the dataset. However, the vocab is created before the ngrams are added. `cntr = Counter(tokens_list) train_vocab = gluonnlp.Vocab(cntr)`

Reply:
right, we can add this to the vocab, but since most obvious case doesnt involve running with additional n-grams, we can keep vocab separate and add n-gram indices this way.

Reply:
Though you create virtual indices for ngrams, the size of embedding matrix is limited to `len(train_vocab)`. See your code `net = FastTextClassificationModel(len(train_vocab), embedding_dim, num_classes)`. When you use `embeddings = self.embedding(doc)` for a lookup in the embedding matrix, it will return the embedding vector of unknown token whenever the query index is larger or equal to `len(train_vocab)`.

Reply:
Very good point, I totally missed it. Thnx for pointing it.
Making this change actually made numbers better than those reported in the paper

Reply:
In the paper, they partition the ngrams into bins, where the ngrams in the same bin share an embedding vector. This method dramatically reduces the memory cost. I think this can be left as a future feature to be included.

Reply:
I opened #589 to track the issue. @sravanbabuiitm do you intend to address this in a follow-up PR?